### PR TITLE
Simplify port handling in butterfly server

### PR DESCRIPTION
**Pull request by krnowak**
The butterfly server allowed to specify a separate port number for
swim connections and a separate port for gossip connections. The only
serious user of the butterfly server, the supervisor, was passing the
same number in for both. It works, because even if butterfly is still
opening two ports, one is a UDP port and the other is TCP port.

Another reason for this change is to simplify the discovery of the
exposed address and port for the supervisor being behind a NAT. This
feature may come in future work revolving around the spanning ring.

Signed-off-by: Krzesimir Nowak <krzesimir@kinvolk.io>

This is conflicting with another PR (#10) I filed today, but that doesn't matter - this is mostly a strawman PR that I want to propose to upstream. Whichever gets merged first - it will become a base of the following PR.

### CODE REVIEW

**Comment by asymmetric** [comment]
@krnowak waiting for tests to be green before reviewing 👅

**Comment by krnowak** [comment]
rustfmt ffs.

**Comment by asymmetric** [comment]
Could you summarize what the simplification is? I couldn't get that from reading the PR description, sorry :)

**Comment by krnowak** [comment]
Instead of passing two port numbers to the `Server::new` function we pass just one, because Supervisor (the only real-world user of butterfly server) is passing the same number for both.

**Review by asymmetric — APPROVED** [review]

**Comment by zeenix** [comment]
When commit log uses vague verbs like "Simplify" and there are a lot of hunks in the patch, it typically means that patch can/should likely be further divided. I'm not quite sure in this case though but can't some of the renaming be in a separate patch that comes before in preparation for the actual changes?

**Comment by krnowak** [comment]
Maybe, but it's just a proposal, so I won't bother with splitting it. I probably shouldn't have filed an internal PR for it anyway. But rather when the idea is approved.
